src/components/picker/new.tsx

```diff
@@ -0,0 +1,266 @@
+// TODO: deprecate all places where we check if _.isPlainObject
+// TODO: deprecate getItemValue prop
+// TODO: deprecate getItemLabel prop
+// TODO: Add initialValue prop
+// TODO: consider deprecating renderCustomModal prop
+// TODO: deprecate onShow cause it's already supported by passing it in pickerModalProps
+import _ from 'lodash';
+import React, {useMemo, useState, useRef, PropsWithChildren} from 'react';
+import {LayoutChangeEvent} from 'react-native';
+import {
+  Constants,
+  asBaseComponent,
+  forwardRef,
+  ForwardRefInjectedProps,
+  BaseComponentInjectedProps
+} from '../../commons/new';
+import ExpandableOverlay, {ExpandableOverlayProps} from '../../incubator/expandableOverlay';
+// @ts-expect-error
+import {TextField} from '../inputs';
+import TextFieldMigrator from '../textField/TextFieldMigrator';
+// @ts-expect-error
+import NativePicker from './NativePicker';
+import PickerItemsList from './PickerItemsList';
+import PickerItem from './PickerItem';
+import PickerContext from './PickerContext';
+import usePickerSelection from './helpers/usePickerSelection';
+import usePickerLabel from './helpers/usePickerLabel';
+import usePickerSearch from './helpers/usePickerSearch';
+import usePickerMigrationWarnings from './helpers/usePickerMigrationWarnings';
+import {PickerProps, PickerValue, PickerModes, PickerSearchStyle} from './types';
+
+const PICKER_MODES = {
```

Comment:
There is a `PickerModes` enum in types, why hold another one?

Reply:
Removed